app/tests/util/member_personas.py

```diff
@@ -123,3 +138,199 @@ async def bob_and_alice_connection(
+    yoma_client: AcaPyClient,
+    schema_definition: SchemaSendResult,
+) -> CredentialExchange:
+
+    await register_issuer(yoma_client, schema_definition.schema_id)
+
+    # Create a conenction from yoma to bob
+    invitation_response = (
+        await yoma_client.post(
+            "/generic/connections/accept-invitation",
+            json={
+                "invitation": register_bob_multi["multi_use_invitation"]["invitation"]
+            },
+        )
+    ).json()
+
+    bob_connection_records = (
+        await bob_member_client.get(f"/generic/connections")
+    ).json()
+    print(json.dumps(bob_connection_records, indent=2))
+
+    bob_connection_id = bob_connection_records[0]["connection_id"]
+
+    # send a credential to bob
+
+    response = await bob_member_client.get(
+        "/generic/issuer/credentials",
+        params={"connection_id": bob_connection_id},
+    )
+    records = response.json()
+
+    # nothing currently in bob's records
+    assert len(records) == 0
+
+    credential = {
+        "protocol_version": "v1",
+        "connection_id": invitation_response["connection_id"],
+        "schema_id": schema_definition.schema_id,
+        "attributes": {"speed": "10"},
+    }
+
+    # create and send credential offer- issuer
+    response = await yoma_client.post(
+        "/generic/issuer/credentials",
+        json=credential,
+    )
+    response.raise_for_status()
+    response = response.json()
+
+    # give aca-py some time to process
+    time.sleep(3)
+
+    # get cred offer record - holder
+    response = await bob_member_client.get("/generic/issuer/credentials")
+    records = response.json()
+    cred_id = records[0]["credential_id"]
+
+    # send credential request - holder
+    response = await bob_member_client.post(
+        f"/generic/issuer/credentials/{cred_id}/request", json={}
+    )
+
+    # send credential - issuer
+    response = await yoma_client.post(
+        "/generic/issuer/credentials",
+        json=credential,
+    )
+    response.raise_for_status()
+
+    # give aca-py some time to process
+    time.sleep(3)
+
+    response = await bob_member_client.post(
+        f"/generic/issuer/credentials/{cred_id}/store", json={}
+    )
+    assert check_webhook_state(
+        client=bob_member_client,
+        filter_map={"state": "credential-acked"},
+        topic="issue_credential",
+    )
+    return response.json()
+
+
+@pytest.fixture(scope="module")
+async def alice_bob_connect_multi(
+    bob_member_client,
+    alice_member_client,
+    bob_multi_use_invitation,
+) -> BobAliceConnect:
+    invitation = bob_multi_use_invitation["multi_use_invitation"]["invitation"]
+
+    # accept invitation on alice side
+    invitation_response = (
+        await alice_member_client.post(
+            "/generic/connections/accept-invitation",
+            json={"invitation": invitation},
+        )
+    ).json()
+
+    print(f"\n\n\n\n Invitation: {invitation}\n\n\n")
+    time.sleep(10)
```

Comment:
We have nice event listeners now. Sleeps make the tests unnecesarily slow

Reply:
oops sorry yeah left-over from the dark ages